core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1401,6 +1401,12 @@ class ReplicaManager(val config: KafkaConfig,
                     if (partitionState.leader != localBrokerId)
                       topicIdUpdateFollowerPartitions.add(partition)
                     Errors.NONE
+                  case None if logTopicId.isDefined =>
+                    // If we have a topic ID in the log but not in the request, we must have previously had topic IDs but
```

Comment:
Should we log something in this case as well?

Reply:
I'm thinking we should move the other if condition up so that the log message isn't confusing. (ie, if the broker is a leader we go to the other case)